src/software/ai/intent/intent_visitor.h

```diff
@@ -28,9 +25,6 @@ class IntentVisitor
      *
      * @param The Intent to visit
      */
-    virtual void visit(const ChipIntent &chip_intent)                  = 0;
-    virtual void visit(const KickIntent &kick_intent)                  = 0;
-    virtual void visit(const MoveIntent &move_intent)                  = 0;
-    virtual void visit(const SpinningMoveIntent &spinning_move_intent) = 0;
-    virtual void visit(const StopIntent &stop_intent)                  = 0;
+    virtual void visit(const MoveIntent &intent)            = 0;
+    virtual void visit(const DirectPrimitiveIntent &intent) = 0;
```

Comment:
Why don't we allow visiting of the other intents here?

Reply:
The subclasses of `DirectPrimitiveIntent` are completely covered by visiting `DirectPrimitiveIntent`

Reply:
Could we leave a quick comment indicating that here?

Reply:
added comments